packages-exp/auth-compat-exp/src/user.ts

```diff
@@ -22,20 +22,41 @@ import {
   convertConfirmationResult,
   convertCredential
 } from './user_credential';
+import { unwrap, Wrapper } from './wrap';
+
+export class User implements compat.User, Wrapper<externs.User> {
+  // Maintain a map so that there's always a 1:1 mapping between new User and
+  // legacy compat users
+  private static readonly USER_MAP = new Map<externs.User, User>();
```

Comment:
should this be a weak map? what happens when a user is deleted?

Reply:
Switched to weak map.

If the developer's code calls `user.delete()`, but retains a reference to the user object, they will still be able to see all the fields on the user-- we don't do any cleanup. So this implementation is still consistent with that